### DISCUSSION TITLE
What is the equivalent of the proxy chain command in msf console?

### DISCUSSION THREAD

**Discussion by flamecopper** [rapid7/metasploit-framework#16272] [General]
Proxy Chaining Scanning
ssh -D 9050 xxx@172.25.xxxx.100 -p60000
/etc/proxychains -> socks4 127.0.0.1 9050
proxychains nmap 172.17.0.1
ProxyChains-3.1 (http://proxychains.sf.net)
Starting Nmap 7.92 ( https://nmap.org ) at 2022-03-04 20:54 +08
|S-chain|-<>-127.0.0.1:9050-<><>-172.17.0.1:80-<><>-OK
|S-chain|-<>-127.0.0.1:9050-<><>-172.17.0.1:21-<--timeout
|S-chain|-<>-127.0.0.1:9050-<><>-172.17.0.1:143-<--timeout

**Comment by smcintyre-r7** [comment]
The equivalent would be to obtain a Meterpreter session and then route all of your traffic through it. I don't believe we have the ability to use an upstream socks proxy for everything, but individual modules do often have HttpProxy options that you can specify.
